src/TTextEdit.cpp

```diff
@@ -1029,10 +1029,11 @@ void TTextEdit::mouseMoveEvent(QMouseEvent* event)
 
 // Returns the index into the relevant TBuffer::lineBuffer of the FIRST QChar
 // of the grapheme under the mouse - it ALSO returns zero (which will probably
-// NOT be a valid index) if there is no valid index to return; it might be
-// worth changing that to -1 but that will probably require revision to the
-// caller(s):
-int TTextEdit::convertMouseXToBufferX(const int mouseX, const int lineNumber) const
+// NOT be a valid index) if there is no valid index to retur.
+// If a pointer to a boolean is provided as a third argument then it will
+// be set to true if the mouse is positioned over a visible time stamp
+// and left unchanged otherwise.
+int TTextEdit::convertMouseXToBufferX(const int mouseX, const int lineNumber, bool* isOverTimeStamp) const
```

Comment:
Pointers to basic types aren't necessary anymore - remember we can do normal multi-returns from C++ now: https://skebanga.github.io/structured-bindings :)

Reply:
It was done like that because I only needed the detail in one place but the method is used several times so I only had to revise the caller in that one place and the other uses could continue unmodified.

Reply:
That seems like an odd reason for API design. Not a big deal in this case though!